src/main/resources/templates/bamboo/c/staticCodeAnalysisRuns/1_run_static_code_analysis.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# ------------------------------
+# Task Description:
+# Build and run all tests if the compilation succeeds
+# ------------------------------
+
+# Artemis will look for SCA reports in target, specified in BambooBuildPlanService.java in Artemis
+sudo rm -rf target
+sudo mkdir target
```

Comment:
Just for clarification as I don’t know about Bamboo internals: e.g. the swift template [does not use sudo](https://github.com/ls1intum/Artemis/blob/2db68939b1d72ac22fd4958454efd08d06df3a7e/src/main/resources/templates/bamboo/swift/staticCodeAnalysisRuns/1_run_static_code_analysis.sh#L4) for the same task.
I’m assuming it is needed in this case here as you specifically added it in a later commit after testing.

* Is there some difference in the images used that causes the need for `sudo`?
* Are the files of the other pipeline types cleaned up properly without? (Definitely not in scope for this PR, but maybe worth investigating separately.)

Reply:
- This is needed here since the [Dockerfile](https://github.com/ls1intum/artemis-c-docker/blob/main/Dockerfile) creates a root user called `artemis_user`, which does not have full permissions on the Bamboo working directory.  Removing `sudo` would therefore not work with the current setup.  Similar to why `sudo` is also used [here](https://github.com/ls1intum/Artemis/blob/2db68939b1d72ac22fd4958454efd08d06df3a7e/src/main/resources/templates/bamboo/c/regularRuns/1_setup_the_build_environment.sh). The swift `Dockerfile` probably uses just the `root` user by default.
- I assume you mean the clean up for regular builds: All folders need to get removed by using `sudo` like it's done at the moment, however, that should result in a proper clean up. At least in the sense of what commonly seems to be meant by "cleaning up the working directory".

On a further note:
-  Nonetheless, there is still something not quite right with the build plan. There seems to be a permission error with the Bamboo internal clean up script that is executed after the final tasks. This problem persists with the current setup in general though, and is not related to this PR. This should not have any negative effect at the moment.
- It might be caused by [this](https://confluence.atlassian.com/bamkb/docker-runner-fails-with-permission-denied-errors-in-build-log-1031279813.html) issue, since Fedora is used for the image. From my understanding Bamboo would clean up the working dir automatically, so deleting the folders in a dedicated task could potentially be avoided. Although the current setup works, it is not perfectly clear to me what Bamboo does internally after the final tasks...

Reply:
Okay, thank you very much for the thorough explanation.